code/game/objects/effects/effect_system/foam.dm

```diff
@@ -39,8 +39,7 @@
 			M.updateicon()
 
 		flick("[icon_state]-disolve", src)
-		sleep(5)
-		qdel(src)
+		QDEL_IN(src, 5)
```

Comment:
```suggestion
		QDEL_IN(src, 0.5 SECONDS)
```

Reply:
length of the animation in frames

Reply:
```suggestion
		QDEL_IN(src, 5) //Animation is 5 frames long.
```

Reply:
its sort of self-evident given the line immediately before it is flick() and its a fairly standard pattern in the code base to use flick() followed by a delay for it to animate

Reply:
Would it be negative to add it?

Reply:
CHECK_BITFIELD